enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,551 @@
+  * _We could trick CI to build `aws-ebs-csi-driver-operator` image from the new repo and use it in presubmit
+    tests there. But we can't promote it anywhere, as it would overwrite the old image from
+    github.com/openshift/aws-ebs-csi-driver-operator. Some experiments are needed here._
+
+  The openshift/csi-operator repository should look like this at this point:
+  ```text
+  ├── assets          # Common assets for all CSI drivers (empty now)
+  ├── cmd             # All commands for all CSI driver operators (empty now)
+  ├── pkg             # All common code for all CSI driver operators (empty now)
+  ├── test            # All tests for all CSI driver operators (empty now)
+  ├── vendor          # Global vendor dir (empty now)
+  ├── legacy          # "Old" CSI driver operators that just merged here.
+  │   └── aws-ebs-csi-driver-operator
+  │       ├── assets
+  │       │   ├── assets.go
+  │       │   ├── cabundle_cm.yaml
+  │       │   └── ... # all other assets
+  │       ├── cmd
+  │       │   └── aws-ebs-csi-driver-operator
+  │       │       └── main.go
+  │       ├── pkg
+  │       │   ├── dependencymagnet
+  │       │   │   └── dependencymagnet.go
+  │       │   ├── operator
+  │       │   │   ├── starter.go
+  │       │   │   ├── starter_test.go
+  │       │   │   ├── storageclasshook.go
+  │       │   │   └── storageclasshook_test.go
+  │       │   └── version
+  │       │       └── version.go
+  │       ├── test
+  │       │   └── e2e
+  │       │       └── manifest.yaml
+  │       ├── vendor
+  │       │   └── ... # all packages vendored by the old operator
+  │       ├── go.mod  # go.mod + go.sum from the old operator
+  │       └── go.sum
+  ├── Dockerfile.aws-ebs
+  └── Dockerfile.aws-ebs.test
+  ```
+  `Dockerfile.aws-ebs` + `Dockerfile.aws-ebs.test` will be in the root directory, so we can re-use it in the next step.
+  They will build the operator + its test image from the `legacy/` directory at this point.
+
+* We coordinate with ART to switch building of `ose-aws-ebs-csi-driver-operator-container` image from
+  github.com/openshift/aws-ebs-csi-driver-operator / `Dockerfile.rhel7` to
+  github.com/openshift/csi-operator / `Dockerfile.aws-ebs` in a semi-atomic way.
+  * With a quick pre-merge test in CI, if possible.
+  * Goal: nightlies should be green all the time.
+  * TBD: exact procedure & tickets to file. Current high level idea:
+    1. PR against openshift/release to stop promoting the CI operator image
+       from `openshift/aws-ebs-csi-driver-operator` jobs and start promoting it from `openshift/csi-operator`.
+    2. PR against openshift/ocp-build-data to switch the source github repo + Dockerfile
+       for `ose-aws-ebs-csi-driver-operator-container` image.
+    3. Somehow coordinated merge of these two.
+       _Can we merge 1. before 2. to see if / how it breaks CI builds? We could be able to revert back to the working
```

Comment:
We can test production builds under 2. independent of 1. We can coordinate timings of merges. Reverts in production build config can be done without too much ceremony.